### PR TITLE
Dashboards: Fix scroll position not being restored when leaving panel edit

### LINKED ISSUE #83783: Dashboards: Scroll position not restored when leaving Edit Panel
On a long Dashboard, when you leave the Edit Panel view your scroll position is supposed to be restored.

Updating react-virtualised-auto-sizer broke this https://github.com/grafana/grafana/pull/80596. From initial investigation, it seems that on the initial render AutoSizer (inside DashboardGrid) does not render it's children in the DOM, so when we attempt to scroll down the page, there's nothing to scroll.

It doesn't appear as if AutoSizer is actually used in a useful way here - it looks like its only used to get the available width of the page in order to pass into ReactGridLayout. It can probably be removed in favor of just measuring the div directly.

### PR DESCRIPTION
**Pull request by joshhunt**
Regression introduced in https://github.com/grafana/grafana/pull/80596 where the Dashboard scroll position would not be restored when leaving panel edit view.

The issue was caused by newer version of AutoSizer not rendering children in the DOM on initial render, so the page wouldn't have anywhere to scroll to. This PR sidesteps the issue by just not using AutoSizer - it's only used to get the width of the container anyway.

Screenshot 2024-03-01 at 5 13 07 pm

Fixes https://github.com/grafana/grafana/issues/83783

### CODE REVIEW

**Review by oscarkilhed — APPROVED** [review]

**Comment by torkelo** [comment]
@joshhunt an escalation for this https://github.com/grafana/support-escalations/issues/10226 

not sure if this is critical enough for a backport

**Comment by joshhunt** [comment]
I don't think this is critical enough to backport to be honest, especially given I think this is just the first report of the bug. Plus I'm not _super_ comfortable backporting the specific changes.

Up to ya'll.

File: public/app/features/dashboard/dashgrid/DashboardGrid.tsx [hunk 1]
```diff
@@ -293,69 +294,69 @@ export class DashboardGrid extends PureComponent<Props, State> {
...
       <div
+        ref={this.onMeasureRef}
         style={{
           flex: '1 1 auto',
           position: 'relative',
           zIndex: 1,
           display: this.props.editPanel ? 'none' : undefined,
         }}
       >
-        <AutoSizer disableHeight>
```
Comment (torkelo) [inline public/app/features/dashboard/dashgrid/DashboardGrid.tsx:319]: @joshhunt this change to useMeasure breaks resizing a window and having the dashboard response to change, for that you need the container to have position absolute, that is at least the only solution I have found

Comment (torkelo) [reply public/app/features/dashboard/dashgrid/DashboardGrid.tsx:319]: hm.. might have been something with a branch I was on, strange, can't replicate in main

**Pull request merged**